src/app/static/src/app/components/genericChart/GenericChart.vue

```diff
@@ -198,22 +200,28 @@
             DownloadIndicator,
             VueFeather,
             DataSource,
-            Filters,
+            FiltersComp,
             Plotly,
             GenericChartTable,
             ErrorAlert,
             LoadingSpinner
         },
-        data: function() {
+        data: function(): Data {
             const chart = this.metadata[this.chartId];
             const dataSourceSelections = chart.dataSelectors.dataSources
-                .reduce((running: Record<string, DataSourceSelections>, dataSource: DataSourceConfig) => ({
-                    ...running,
-                    [dataSource.id]: {
-                        datasetId: this.availableDatasetIds.find(id => id === dataSource.datasetId) || this.availableDatasetIds[0],
-                        selectedFilterOptions: null
-                    }
-                }), {});
+            .reduce((running: Record<string, DataSourceSelections>, dataSource: DataSourceConfig) => Object.assign(running, {
+                [dataSource.id]: {
+                    datasetId: this.availableDatasetIds.find(id => id === dataSource.datasetId) || this.availableDatasetIds[0],
+                    selectedFilterOptions: null
+                }
+            }), {})
+            // .reduce((running: Record<string, DataSourceSelections>, dataSource: DataSourceConfig) => ({
+            //         ...running,
+            //         [dataSource.id]: {
+            //             datasetId: this.availableDatasetIds.find(id => id === dataSource.datasetId) || this.availableDatasetIds[0],
+            //             selectedFilterOptions: null
+            //         }
+            //     }), {});
```

Comment:
Can this be sorted ?

Probably creeped in

Reply:
removed